com.unity.ml-agents/Runtime/Grpc/RpcCommunicator.cs

```diff
@@ -325,26 +328,33 @@ void SendBatchedMessageHelper()
                 }
 
                 var agentActions = rlInput.AgentActions[brainName].ToAgentActionList();
-                var numAgents = m_ActionCallbacks[brainName].Count;
-                var agentActionDict = new Dictionary<int, AgentAction>(numAgents);
-                m_LastActionsReceived[brainName] = agentActionDict;
+                var numAgents = m_ActionId[brainName].Count;
                 for (var i = 0; i < numAgents; i++)
                 {
                     var agentAction = agentActions[i];
-                    var agentId = m_ActionCallbacks[brainName][i].AgentId;
-                    agentActionDict[agentId] = agentAction;
-                    m_ActionCallbacks[brainName][i].Callback.Invoke(agentAction);
+                    var agentId = m_ActionId[brainName][i];
+                    if (m_LastActionsReceived[brainName].ContainsKey(agentId))
+                    {
+                        m_LastActionsReceived[brainName][agentId] = agentAction.vectorActions;
+                    }
                 }
             }
-            foreach (var brainName in m_ActionCallbacks.Keys)
+            foreach (var brainName in m_ActionId.Keys)
             {
-                m_ActionCallbacks[brainName].Clear();
+                m_ActionId[brainName].Clear();
             }
         }
 
-        public Dictionary<int, AgentAction> GetActions(string key)
+        public float[] GetActions(string key, int agentId)
```

Comment:
`key` -> `brainName` for consistency. Or better yet, `behaviorName` and replace `brainName` above.